.github/workflows/standard_spa_pr.yml

```diff
@@ -98,24 +115,33 @@ jobs:
 
   deploy_magic_url:
     name: Deploy Magic URL
-    runs-on: jupiterone-dev
-    needs: cache-yarn
+    runs-on: ${{ inputs.runs_on }}
+    timeout-minutes: 15
     if: ${{ inputs.use_magic_url }}
     permissions:
       id-token: write
       contents: read
       pull-requests: write
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: Fetch Yarn Cache
-        uses: actions/cache@v2
-        id: yarn-cache
+      - uses: actions/checkout@v3
+      # This should be part of the runner
+      ## Install yarn START
+      - uses: actions/setup-node@v3
+        if: ${{ inputs.install_yarn }}
         with:
-          path: |
-            ${{ steps.yarn-cache-dir-path.outputs.dir }}
-            **/node_modules
-          key: cache-yarn-${{ hashFiles('**/yarn.lock') }}
+          node-version: 16
+      - run: npm install -g yarn
+        if: ${{ inputs.install_yarn }}
+      ## Install yarn STOP
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: "yarn"
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      # Only allow production deps be installed in the final build!
+      - run: yarn --frozen-lockfile --production --ignore-optional
```

Comment:
This is the same build that we will want in our true production bundles. So if this fails, we will know that deployment from `main` will fail.